uc-crux-llvm/test/Test.hs

```diff
@@ -357,11 +363,12 @@ inFileTests =
         ("write_to_null.c", [("write_to_null", hasBugs)]),
         ("branch.c", [("branch", isSafe mempty)]),
         ("compare_to_null.c", [("compare_to_null", isSafe mempty)]),
+        ("extern_void_function.c", [("extern_void_function", isSafe (skipOverride "do_stuff"))]),
         -- This override needs refinement; the following should be safe with the
         -- precondition that the argument pointer is valid.
-        ("getenv_arg.c", [("getenv_arg", isSafe (usesOverride "getenv"))]),
-        ("getenv_const.c", [("getenv_const", isSafe (usesOverride "getenv"))]),
-        ("gethostname_const_len.c", [("gethostname_const_len", isSafe (usesOverride "gethostname"))]),
+        ("getenv_arg.c", [("getenv_arg", isSafe (unsoundOverride "getenv"))]),
+        ("getenv_const.c", [("getenv_const", isSafe (unsoundOverride "getenv"))]),
```

Comment:
The code here is mostly complete (insofar as the context of this PR, there are TODOs around the extension to non-void functions that I want to tackle soon), but these two tests have started failing for reasons I don't yet understand.
```
    getenv_arg.c:                                                      FAIL (0.06s)
      test/Test.hs:202:
      expected: Unsoundness {unsoundOverridesUsed = fromList [UnsoundOverrideName {getUnsoundOverrideName = "getenv"}], unsoundSkipOverridesUsed = fromList []}
       but got: Unsoundness {unsoundOverridesUsed = fromList [], unsoundSkipOverridesUsed = fromList []}
    getenv_const.c:                                                    FAIL (0.06s)
      test/Test.hs:202:
      expected: Unsoundness {unsoundOverridesUsed = fromList [UnsoundOverrideName {getUnsoundOverrideName = "getenv"}], unsoundSkipOverridesUsed = fromList []}
       but got: Unsoundness {unsoundOverridesUsed = fromList [], unsoundSkipOverridesUsed = fromList []}
```
The strange thing here is that I didn't really change anything about how the unsound overries operate...

Reply:
If I remove the call to `registerUnsoundSkipOverrides`, then these tests pass.

Reply:
It's not that a different override is getting registered for `getenv`... With this diff:
```diff
diff --git a/uc-crux-llvm/src/UCCrux/LLVM/Overrides/Skip.hs b/uc-crux-llvm/src/UCCrux/LLVM/Overrides/Skip.hs
index 9eb03c9d..05e3a5a6 100644
--- a/uc-crux-llvm/src/UCCrux/LLVM/Overrides/Skip.hs
+++ b/uc-crux-llvm/src/UCCrux/LLVM/Overrides/Skip.hs
@@ -12,6 +12,7 @@ Stability    : provisional
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE PolyKinds #-}
 {-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TupleSections #-}
 
 module UCCrux.LLVM.Overrides.Skip
   ( SkipOverrideName (..),
@@ -93,9 +94,12 @@ registerUnsoundSkipOverrides proxy llvmModule mtrans usedRef decls =
               (handleMapToHandles (fnBindings binds))
     let overrides =
           mapMaybe
-            (createSkipOverride proxy usedRef)
+            (\dec -> (dec,) <$>  createSkipOverride proxy usedRef dec)
             (filter ((`Set.notMember` alreadyDefined) . declName) decls)
-    register_llvm_overrides llvmModule [] overrides llvmCtx
+    mapM_
+      (\dec -> liftIO (putStrLn ("Skipping " ++ Text.unpack (declName dec))))
+      (map fst overrides)
+    register_llvm_overrides llvmModule [] (map snd overrides) llvmCtx
```
I only see `llvm.dbg.value` getting skipped.

Maybe it's that I'm calling `register_llvm_overrides` twice? I'll look deeper into that function.

Reply:
Yep, `register_llvm_overrides` can only be called once. Fixed!